### PR TITLE
Add tutorial for a custom envelope and a C service.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: docs/tutorials/misc/c-service.mdx [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+#include <string.h>
+
+toit_msg_request_handle_t current_handle;
+
+static void on_qrcode_generated(esp_qrcode_handle_t qrcode) {
+  int size = esp_qrcode_get_size(qrcode);
+  // 1 byte to store the size of the QR code.
+  // Then ceil(size * size / 8) bytes to store the data.
+  int data_size = 1 + (size * size + 7) / 8;
+  uint8_t* data = toit_malloc(data_size);
```
Comment (kasperl) [inline docs/tutorials/misc/c-service.mdx:222]: Use toit_calloc?

**Review by kasperl — COMMENTED** [review]
Looks good to me.

File: docs/tutorials/misc/c-service.mdx (hunk 1 above)
Comment (floitsch) [reply docs/tutorials/misc/c-service.mdx:222]: new PR already pending.

**Pull request merged**